### PR TITLE
fix mout vulnerability by replacing credential package with credentials

### PR DESCRIPTION
**Pull request by ETLaurent**
## Summary

* Replace [`credential`](https://www.npmjs.com/package/credential) package with [`credentials`](https://www.npmjs.com/package/credentials) to fix the [`mout` Prototype Pollution vulnerability](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-7792).

## What are the specific steps to test this change?

- link apostrophe (branch 2.0) into a project ([apostrophe-enterprise-testbed](https://github.com/apostrophecms/apostrophe-enterprise-testbed) for instance)
- test login with a correct password -> should log in
- test login with a wrong password -> should not log in

⚠️ former users that were created before this code change (i.e with the credential package) should still be able to log in with their password.

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [x] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by boutell — DISMISSED** [review]

File: lib/modules/apostrophe-users/index.js [hunk 1]
```diff
@@ -82,7 +82,6 @@ module.exports = {
   slugPrefix: 'user-',
 
   afterConstruct: function(self, callback) {
-    self.initializeCredential();
```
[the remark this answers is not in the document]
Comment (ETLaurent) [reply lib/modules/apostrophe-users/index.js:85]: Hmm you're right, I did not think of that...
Ok!

[the remark this answers is not in the document]
Comment (ETLaurent) [reply lib/modules/apostrophe-users/index.js:85]: This "all public" functions are making all lot of BCs...

File: test/users.js [hunk 2]
```diff
@@ -103,6 +103,38 @@ describe('Users', function() {
...
+        // A password hash that were generated by the former credential package:
+        var oldPasswordHashSimulated =
+          '{"hash":"HKBAyPWKKnKnXzF0yflRUEeeJZk1njKaX3IqT6Ml056OdMWIsDRqfJeHCqxI3jA9HEFNzuPEhw0m98dA8ju8xRpj","salt":"P2X4+Ex0rrHSPBRv0TCGOTqXmuT2JDspNLc/0Uln6jcZWACUpgBz+DDpfP9DFZcPG9cMlwMaHEKw3MVq02af8RSn","keyLength":66,"hashMethod":"pbkdf2","iterations":2853010}';
+
+        apos.users.safe.update({ username: 'olduser' }, { $set: { passwordHash: oldPasswordHashSimulated } }, function() {
+          apos.users.verifyPassword(user, 'passwordThatThroughOldCredentialPackageHashing', function(err) {
+            assert(!err);
+            apos.users.safe.remove({ username: 'olduser' }, function() {
+              done();
+            });
```
Comment (ETLaurent) [inline test/users.js:131]: remove to avoid polluting next tests

File: test/users.js [hunk 3]
```diff
@@ -103,6 +103,38 @@ describe('Users', function() {
...
+    user.email = 'old@user.com';
+
+    apos.users.insert(req, user, async function(err) {
+      assert(!err);
+      apos.users.safe.findOne({ username: 'olduser' }, function(err) {
+        assert(!err);
+
+        // A password hash that were generated by the former credential package:
+        var oldPasswordHashSimulated =
+          '{"hash":"HKBAyPWKKnKnXzF0yflRUEeeJZk1njKaX3IqT6Ml056OdMWIsDRqfJeHCqxI3jA9HEFNzuPEhw0m98dA8ju8xRpj","salt":"P2X4+Ex0rrHSPBRv0TCGOTqXmuT2JDspNLc/0Uln6jcZWACUpgBz+DDpfP9DFZcPG9cMlwMaHEKw3MVq02af8RSn","keyLength":66,"hashMethod":"pbkdf2","iterations":2853010}';
```
Comment (ETLaurent) [inline test/users.js:124]: If there is a better solution I'll take it

File: test/users.js [hunk 4]
```diff
@@ -103,6 +103,38 @@ describe('Users', function() {
...
 
+  it('should verify a user password created with former credential package', function(done) {
+    var req = apos.tasks.getReq();
+    var user = apos.users.newInstance();
+
+    user.firstName = 'Old';
+    user.lastName = 'User';
+    user.title = 'Old User';
+    user.username = 'olduser';
+    user.password = 'passwordThatThroughOldCredentialPackageHashing';
```
Comment (ETLaurent) [inline test/users.js:114]: `passwordThatThroughOldCredentialPackageHashing` -> Went _Through_, forgot that.. but we're in a test, and in A2, so I guess it's ok

**Review by boutell — APPROVED** [review]

**Pull request merged**